elasticdl/python/master/master.py

```diff
@@ -37,13 +38,21 @@ def _make_task_dispatcher(
     prediction_data,
     records_per_task,
     num_epochs,
+    data_reader_params,
 ):
     # TODO: Support any subclasses of `AbstractDataReader`
     # and support passing specified parameters to the constructor
     def _maybe_create_shards(data_origin):
+        wkargs = get_dict_from_params_str(data_reader_params)
+        if wkargs and "partition" in wkargs:
+            partition = wkargs["partition"]
+        else:
```

Comment:
wkargs.get(key, default_value) is probably what you wanted?

Reply:
Rewrote.